ttnn/cpp/ttnn/operations/data_movement/split/device/split_op.cpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "split_op.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/copy/copy_op.hpp"
+
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/common/constants.hpp"
+
+#include "split_program_factory.hpp"
+using namespace tt::constants;
+
+
+namespace ttnn::operations::data_movement {
+
+void Split::validate(const std::vector<Tensor> &input_tensors) const {
+    const auto &input_tensor = input_tensors.at(0);
+    tt::tt_metal::Buffer *in0_buffer = input_tensor.buffer();
+
+    tt::DataFormat cb_data_format = tt::DataFormat::Bfp8_b;
+    if (input_tensor.get_dtype() == tt::tt_metal::DataType::BFLOAT16) {
+        cb_data_format = tt::DataFormat::Float16_b;
+    }
+
+    uint32_t single_tile_size = tt::tt_metal::detail::TileSize(cb_data_format);
+    TT_FATAL(this->dim == 3 || this->dim == 2, "split is possible along dim 2 or 3 only");
+    TT_FATAL(in0_buffer->size() % single_tile_size == 0);
```

Comment:
Could you add comment about buffer size should be divisible by single_tile_size.
If you think it is internal check I suggest to use TT_ASSERT then.

Reply:
I don't need it (already checking layout), I'll get rid of it. I'll clean this up a bit. I copy pasted from the old tt_lib vsn of it